src/Models/EmailLink.php

```diff
@@ -55,4 +57,11 @@ public function getMenuTitle(): string
     {
         return _t(__CLASS__ . '.LINKLABEL', 'Link to email address');
     }
+
+    public function getCMSCompositeValidator(): CompositeValidator
+    {
+        $validator = parent::getCMSCompositeValidator();
+        $validator->addValidator(RequiredFields::create(['Email']));
```

Comment:
```suggestion
        $requiredFieldsValidators = $validator->getValidatorsByType(RequiredFields::class);
        if (empty($requiredFieldsValidators)) {
            $requiredFieldsValidator = RequiredFields::create(['Email']);
        } else {
            $requiredFieldsValidator = $requiredFieldsValidators[array_key_first($requiredFieldsValidators)];
        }
        $validator->addValidator($requiredFieldsValidator);
```
An alternative approach would be to add a `getOrCreateValidatorOfType()` method to `CompositeValidator` (similar to [`FieldList::findOrMakeTab()`](https://api.silverstripe.org/5/SilverStripe/Forms/FieldList.html#method_findOrMakeTab)) - or at least a `getValidatorByType()` (similar to [`GridFieldConfig::getComponentByType()`](https://api.silverstripe.org/5/SilverStripe/Forms/GridField/GridFieldConfig.html#method_getComponentByType))

You don't have to do this, but it does make the validator tidier and easier to work with in subclasses or extensions. I won't block merging if you don't make this change.

If you do make this change, please also make it in the other `getCMSCompositeValidator()` implementations.

Reply:
Leaving as is